src/Microsoft.ML.CodeGenerator/Templates/Azure/Console/AzureModelBuilder.cs

```diff
@@ -0,0 +1,430 @@
+            this.Write(this.ToStringHelper.ToStringWithCulture(Path));
+            this.Write("\";\r\n        private static string MLNET_MODEL = @\"");
+            this.Write(this.ToStringHelper.ToStringWithCulture(MLNetModelpath));
+            this.Write("\";\r\n\t\tprivate static string ONNX_MODEL = @\"");
+            this.Write(this.ToStringHelper.ToStringWithCulture(OnnxModelPath));
+            this.Write(@""";
+       
+	    // Create MLContext to be shared across the model creation workflow objects 
+        // Set a random seed for repeatable/deterministic results across multiple trainings.
+        private static MLContext mlContext = new MLContext(seed: 1);
+
+		// Training in Azure produces an ONNX model; this method demonstrates creating an ML.NET model (MLModel.zip) from the ONNX model (bestModel.onnx), which you can then use with the ConsumeModel() method to make predictions
+        public static void CreateMLNetModelFromOnnx()
+        {
+            // Load data
+            IDataView inputDataView = mlContext.Data.LoadFromTextFile<ModelInput>(
+                                            path: TRAIN_DATA_FILEPATH,
+                                            hasHeader : ");
+            this.Write(this.ToStringHelper.ToStringWithCulture(HasHeader.ToString().ToLowerInvariant()));
+            this.Write(",\r\n                                            separatorChar : \'");
+            this.Write(this.ToStringHelper.ToStringWithCulture(Regex.Escape(Separator.ToString())));
+            this.Write("\',\r\n                                            allowQuoting : ");
+            this.Write(this.ToStringHelper.ToStringWithCulture(AllowQuoting.ToString().ToLowerInvariant()));
+            this.Write(",\r\n                                            allowSparse: ");
+            this.Write(this.ToStringHelper.ToStringWithCulture(AllowSparse.ToString().ToLowerInvariant()));
+            this.Write(@");
+
+            // Create an ML.NET pipeline to score using the ONNX model
+			// Notice that this pipeline is not trainable because it only contains transformers
+            IEstimator<ITransformer> pipeline = BuildPipeline(mlContext);
+
+			// Create ML.NET model from pipeline
+			ITransformer mlModel = pipeline.Fit(inputDataView);
+
+            // Save model
+            SaveModel(mlContext, mlModel, MLNET_MODEL, inputDataView.Schema);
+        }
+
+        public static IEstimator<ITransformer> BuildPipeline(MLContext mlContext)
+        {
+");
+ if(PreTrainerTransforms.Count >0 ) {
+            this.Write("            // Data process configuration with pipeline data transformations to:\r" +
+                    "\n            // 1. Score using provided onnx model\r\n            // 2. Map scores" +
+                    " to labels to make model output easier to understand and use\r\n            var pi" +
+                    "peline = ");
+ for(int i=0;i<PreTrainerTransforms.Count;i++) 
```

Comment:
Nit: if it's not too hard, it'd be nice to fix the formatting in the generated `CreateMLNetModelFromOnnx()` function.

Reply:
All generated code will be formatted before written to file. So the space will be added

## Resolved